modules/clientnotify.cpp

```diff
@@ -76,18 +89,37 @@ class CClientNotifyMod : public CModule {
 
     void OnClientLogin() override {
         CString sRemoteIP = GetClient()->GetRemoteIP();
-        if (!m_bNewOnly ||
-            m_sClientsSeen.find(sRemoteIP) == m_sClientsSeen.end()) {
+        CString sRemoteClientID = GetClient()->GetIdentifier();
+
+        CString& sClientNameMessage = sRemoteIP;
+        if (m_bNotifyOnNewClientID and sRemoteClientID != "") {
+            sClientNameMessage = sRemoteClientID;
+        }
+
+        auto sendLoginNotification = [&]() {
             SendNotification(t_p("<This message is impossible for 1 client>",
-                                 "Another client authenticated as your user. "
-                                 "Use the 'ListClients' command to see all {1} "
+                                 "Another client ({1}) authenticated as your user. "
+                                 "Use the 'ListClients' command to see all {2} "
                                  "clients.",
                                  GetUser()->GetAllClients().size())(
-                GetUser()->GetAllClients().size()));
-
-            // the set<> will automatically disregard duplicates:
-            m_sClientsSeen.insert(sRemoteIP);
+                sClientNameMessage, GetUser()->GetAllClients().size()));
+        };
+
+        if (m_bNewOnly) {
+            // see if we actually got a new client
+            // TODO: replace setName.find(...) == setName.end() with !setName.contains() once ZNC uses C++20
```

Comment:
Yeah, probably it's time to bump the C++ version further. C++20 is too early to use though

Reply:
Do you create an Issue for this or should I?